context/app/static/js/components/workspaces/utils.js

```diff
@@ -41,4 +41,22 @@ async function startJob({ workspaceId, workspacesEndpoint, workspacesToken }) {
   });
 }
 
-export { createNotebookWorkspace, startJob };
+function mergeJobsIntoWorkspaces(jobs, workspaces) {
+  const wsIdToJobs = {};
+  jobs.forEach((job) => {
+    const { workspace_id } = job;
+    if (!(workspace_id in wsIdToJobs)) {
+      wsIdToJobs[workspace_id] = [];
+    }
+    wsIdToJobs[workspace_id].push(job);
+  });
+
```

Comment:
```suggestion
  jobs.reduce((acc, job) => {
      const { workspace_id } = job;
      if(!(workspace_id in acc)){
          acc[workspace_id] = [];
      }
      acc[workspace_id] = [...acc[workspace_id], job];
      return acc;
      }, {});

```

Reply:
A slight preference for reduce, but up to you. The indentation on the suggestion seems slightly off.

Reply:
I'd need to do something with the result of the `reduce`, right? It's not my usual idiom, but I'm wondering if there are arguments for it beyond:
- No external scratch variable.
- Favoring functional style over procedural.

Also, wondering about the array construction... the point is to favor creating new objects over mutating existing? I worry that it hurts readability.

Reply:
(Going to fix indentation and merge as is, but happy to learn more about your motivation.)

Reply:
... and I realize I don't see the problem with the indentation, either. Sorry not to get it.